File: android/src/main/java/com/getcapacitor/community/inappreview/InAppReviewPlugin.kt
package com.getcapacitor.community.inappreview

import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "InAppReview")
public class InAppReviewPlugin : Plugin() {
    private val implementation = InAppReview()

    @PluginMethod
    public fun requestReview(call: PluginCall) {
        implementation.requestReview(call, activity)
    }
}

File: ios/Sources/InAppReviewPlugin/InAppReviewPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(InAppReviewPlugin)
public class InAppReviewPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "InAppReviewPlugin"
    public let jsName = "InAppReview"
    public let pluginMethods: [CAPPluginMethod] = [
        .async("requestReview", InAppReviewPlugin.requestReview)
    ]
    private let implementation = InAppReview()

    /// The window scenes are UIKit state: the method runs on the main actor. It resolves once the prompt is requested.
    @MainActor
    func requestReview(_ call: CAPPluginCall) async {
        implementation.requestReview()
    }
}
